### DISCUSSION TITLE
Getting nested subclasses

### DISCUSSION THREAD

**Discussion by Cormanz** [classgraph/classgraph#575] [Q&A]
Let's say I have `Grandparent`

`Grandparent` has two subclasses: `Parent` and `Uncle`
`Parent` has one subclass: `Child`

If I get `Grandparent` from `.getClassInfo(Grandparent.getName())` on a scan, how do I get all the (nested) subclasses? Parent, Uncle, and Child?

**Comment by lukehutch** [comment]
Off the top of my head, I think you just call `.getSubclasses()` on the grandparent. Did you try this?

(To get just the parent you would call `.getSubclasses().directOnly()`.)
